controllers/volumesnapshotbackup_controller.go

```diff
@@ -166,6 +167,8 @@ func (r *VolumeSnapshotBackupReconciler) SetupWithManager(mgr ctrl.Manager) erro
 		Owns(&snapv1.VolumeSnapshotContent{}).
 		Owns(&snapv1.VolumeSnapshot{}).
 		Owns(&v1.PersistentVolumeClaim{}).
+		Owns(&volsyncv1alpha1.ReplicationSource{}).
+		Owns(&volsyncv1alpha1.ReplicationDestination{}).
```

Comment:
Should be [`Watches`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/builder#Builder.Watches) not `Owns`.

Only ReplicationSource is relevant to backup controller?

May want to do the same fore restore controller as well?

Reply:
Oh good call. @kaovilai I don't know about watches vs owns. This conversation came up a lot yesterday. Watches requires a separate event handler and I don't know that its needed. I need to understand why Owns is wrong. Do you guys know why?

Reply:
I don't know if owns is wrong but if there are owner references without blockOwnerDeletion it could cause problems.

Reply:
> Owns defines types of Objects being *generated* by the ControllerManagedBy, and configures the ControllerManagedBy to respond to create / delete / update events by *reconciling the owner object*. This is the equivalent of calling Watches(&source.Kind{Type: <ForType-forInput>}, &handler.EnqueueRequestForOwner{OwnerType: apiType, IsController: true}).

Looks like own is the same as watches and is higher level.. so drop my comment on watches.